src/vendored/format.rs

```diff
@@ -0,0 +1,363 @@
+//! Vendored from format.rs in rustpython-vm (see https://github.com/RustPython).
```

Comment:
Do you mind updating this link to include the Git hash of the commit, and the full path to the file? (That is, the permalink to `format.rs`.)